cypress/support/e2e.ts

```diff
@@ -113,14 +113,39 @@ declare global {
   }
 }
 
-beforeEach(() => {
-  cy.login();
-  cy.setCookie(bannerCookie, "true");
-  cy.setCookie(CY_DISABLE_COMMITS_WELCOME_MODAL, "true");
-  cy.setCookie(CY_DISABLE_NEW_USER_WELCOME_MODAL, "true");
-  cy.setCookie(konamiCookie, "true");
-  cy.setCookie(SLACK_NOTIFICATION_BANNER, "true");
+before(() => {
+  cy.exec("yarn restore-evg").then((result) => {
+    if (result.code !== 0) {
+      throw new Error("EVG DB restoration failed during setup.");
+    }
+  });
 });
 
+(function () {
+  let mutationDispatched: boolean;
+  beforeEach(() => {
+    cy.login();
+    cy.setCookie(bannerCookie, "true");
+    cy.setCookie(CY_DISABLE_COMMITS_WELCOME_MODAL, "true");
+    cy.setCookie(CY_DISABLE_NEW_USER_WELCOME_MODAL, "true");
+    cy.setCookie(konamiCookie, "true");
+    cy.setCookie(SLACK_NOTIFICATION_BANNER, "true");
+    mutationDispatched = false;
+    cy.intercept("POST", "/graphql/query", (req) => {
+      if (req.body.query?.startsWith(MUTATION)) {
```

Comment:
nit. Could we extract this into a function just for readability? 
`isMutation(req)`